extensions/2.0/Vendor/GODOT_single_root/README.md

```diff
@@ -0,0 +1,89 @@
+# GODOT_single_root
+
+## Contributors
+
+- Aaron Franke, Godot Engine.
+
+## Status
+
+Complete
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+
+## Overview
+
+This extension imposes constraints on glTF files to ensure they have a single root node.
+
+Many games engines import glTF files as objects with a single root node. Unity has prefabs with a single root GameObject, and Godot has scenes with a single root node. Since glTF allows defining multiple root nodes, engines will insert the glTF root nodes as children of the "real" root node, which makes it difficult for a glTF file to define the properties of the "real" root node. This is important for things like physics bodies and character controllers, which are defined on the root node, so that all child nodes are moved with the body. Aside from physics, a single root node would avoid an extra node in the tree when round-tripping between glTF and these engines.
+
+The `GODOT_single_root` glTF extension solves this problem by imposing additional constraints to ensure the entire glTF file can be parsed into one of these objects with a single root node. Therefore, all other nodes are either descendants of this node, or are not in the glTF scene node tree. Implementations can detect `GODOT_single_root` and import the single root node as the object's "real" root node in the scene/prefab/etc. The extension contains no data, it only restricts behavior. The restrictions are as follows:
```

Comment:
Does the updated language mean that out-of-scene nodes are silently ignored?

Reply:
Aren't they already silently ignored with base glTF, unless some other extension uses them?

Reply:
^ That _might_ have referred to _skeletons_, but I haven't read the details, so that might be completely off-track. 

Iff it referred to skeletons, then maybe 

> all other nodes are either descendants of this node

could be

> all other _non-skeleton_ nodes are either descendants of this node

or so, but all this is to be confirmed - maybe I mistunderstood something here

Reply:
I don't want to impose a restriction either which way for non-scene nodes. They should behave the same as a vanilla glTF asset.

Reply:
Ack, assuming that non-scene node presence does not lead to creating a synthetic "Scene-for-non-Scene-nodes" root.